ember-server/shared/src/main/scala/org/http4s/ember/server/internal/WebSocketHelpers.scala

```diff
@@ -184,8 +184,8 @@ private[internal] object WebSocketHelpers {
       case x => F.pure(Some(x))
     }
 
-  private def frameToBytes(frame: WebSocketFrame): List[Chunk[Byte]] =
-    nonClientTranscoder.frameToBuffer(frame).toList.map { buffer =>
+  private def frameToBytes(frame: WebSocketFrame): Chunk[Chunk[Byte]] =
+    Chunk.array(nonClientTranscoder.frameToBuffer(frame)).map { buffer =>
```

Comment:
Actually if we just return `Chunk[Byte]` here by concatenating then we don't need to traverse at all. I'm not sure why it's written like this.
https://github.com/http4s/http4s/blob/0fb504c6ffd20602d44591bea4d0e784c98b3c4f/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/WebSocketHelpers.scala#L119-L120

I believe the same issue was raised in https://github.com/http4s/http4s/pull/7059 as well.

Reply:
Hmmm That may be a good idea, doing a single socket.write of a larger chunk. However, that would change the I-O profile, which is deeper than just using another data structure for sequence. Perhaps alternative PR that supersedes this one?

Reply:
Whats intresting is that `nonClientTranscoder.frameToBuffer(` breaks it up on purpose. Now I'm not sure I understand the reason, but we need to in order to do that optimization.

Reply:
> breaks it up on purpose. Now I'm not sure I understand the reason, but we need to in order to do that optimization.

Yes, thanks for saying so. I agree, we should make sure to understand the reason.

Reply:
@ChristopherDavenport @armanbilge I agree that it would help to understand why it is broken up, and that having a single write operation of a single concatenated chunk may be more efficient. 

Nevertheless, this PR does not preclude such a change to be done afterwards. Are there any impediments to merging this one?

Reply:
Nope, none.

Reply:
omg, good new